src/main/java/me/desht/modularrouters/item/module/SenderModule3.java

```diff
@@ -42,7 +43,7 @@ public int getEnergyCost(ItemStack stack) {
     }
 
     @Override
-    protected boolean badDimension(ResourceLocation dimId) {
-        return ModularRouters.getDimensionBlacklist().test(dimId);
+    public boolean canOperateInDimension(ResourceKey<Level> dimension) {
+        return ModularRouters.getDimensionBlacklist().test(dimension.location());
```

Comment:
This needs a negation, I believe? Since you're changing the method from a negative sense ("bad dimension") to a positive sense ("can operate").

Reply:
You're right indeed